lightning/mydump/parquet_parser.go

```diff
@@ -185,14 +195,64 @@ func setDatumValue(d *types.Datum, v reflect.Value) {
 		if v.IsNil() {
 			d.SetNull()
 		} else {
-			setDatumValue(d, v.Elem())
+			setDatumValue(d, v.Elem(), meta)
 		}
 	default:
 		log.L().Fatal("unknown value", zap.Stringer("kind", v.Kind()),
 			zap.String("type", v.Type().Name()), zap.Reflect("value", v.Interface()))
 	}
 }
 
+func abs(v int64) int64 {
+	if v >= 0 {
+		return v
+	}
+	return -v
+}
+
+// when the value type is int32/int64, convert to value to target logical type in tidb
+func setDatumByInt(d *types.Datum, v int64, meta *parquet.SchemaElement) {
+	if meta.ConvertedType == nil {
+		d.SetInt64(v)
+		return
+	}
+	switch *meta.ConvertedType {
+	// decimal
+	case parquet.ConvertedType_DECIMAL:
+		scale := int64(1)
+		for i := 0; i < int(*meta.Scale); i++ {
+			scale *= 10
+		}
+		fmtStr := fmt.Sprintf("%%d.%%0%dd", *meta.Scale)
+		val := fmt.Sprintf(fmtStr, v/scale, abs(v%scale))
```

Comment:
```suggestion
		val := fmt.Sprintf("%d.%0*d", v/scale, *meta.Scale, abs(v%scale))
```

but maybe it's more efficient to just print `fmt.Sprintf("%0*d", *meta.Scale + N, v)` (for a suitable value of N) and then insert a `.` to the resulting string.